### PR TITLE
Implement Drawable::Calc_Physics_Xform_Wheels

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/client/draw/drawable.cpp [hunk 1]
```diff
@@ -2098,3 +2101,295 @@ void Drawable::Set_Stealth_Look(StealthLookType look)
...
+                const Coord3D *direction = Get_Unit_Dir_Vector2D();
+                const Coord3D &prev_accel = physics->Get_Prev_Accel();
+                float diry = -direction->y;
+                float dirx = direction->x;
+                PathfindLayerEnum layer = object->Get_Layer();
+
+                Coord3D n;
+                float height = g_theTerrainLogic->Get_Layer_Height(position->x, position->y, layer, &n, true);
+                float pitch = (n.x * direction->x + n.y * direction->y) * 1.5707964f;
+                float roll = (n.x * diry + n.y * dirx) * 1.5707964f;
```
Comment (xezon) [inline src/game/client/draw/drawable.cpp]: `1.5707964f` is probably half pi, used a few times in this function.

File: src/game/client/draw/drawable.cpp [hunk 2]
```diff
@@ -2098,3 +2101,295 @@ void Drawable::Set_Stealth_Look(StealthLookType look)
...
+                        if (m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset < max_wheel_extension) {
+                            m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset = max_wheel_extension;
+                        }
+                    }
+
+                    float z_mult = 4.0f;
+                    float pitch3 = GameMath::Fabs(xform.m_totalPitch - pitch);
+
+                    if (pitch3 > 0.39269909f) {
+                        z_mult = ((pitch3 - 0.39269909f) * 1.0f + 1.5707964f) / pitch3;
```
Comment (xezon) [inline src/game/client/draw/drawable.cpp]: 0.39269909f is an 8th of PI, or PI / 8

File: src/game/client/draw/drawable.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp [hunk 3]
```diff
@@ -2098,3 +2101,295 @@ void Drawable::Set_Stealth_Look(StealthLookType look)
...
+                        + m_drawableLocoInfo->m_accelerationPitchRate;
+                    m_drawableLocoInfo->m_accelerationPitch =
+                        m_drawableLocoInfo->m_accelerationPitch + m_drawableLocoInfo->m_accelerationPitchRate;
+                    m_drawableLocoInfo->m_accelerationRollRate = -roll_stiffness * m_drawableLocoInfo->m_accelerationRoll
+                        + -roll_damping * m_drawableLocoInfo->m_accelerationRollRate
+                        + m_drawableLocoInfo->m_accelerationRollRate;
+                    m_drawableLocoInfo->m_accelerationRoll =
+                        m_drawableLocoInfo->m_accelerationRoll + m_drawableLocoInfo->m_accelerationRollRate;
+                    xform.m_totalPitch = m_drawableLocoInfo->m_pitch + m_drawableLocoInfo->m_accelerationPitch;
+                    xform.m_pitchRate = m_drawableLocoInfo->m_roll + m_drawableLocoInfo->m_accelerationRoll;
```
Comment (tomsons26) [inline src/game/client/draw/drawable.cpp]: should be xform.m_totalRoll =

File: src/game/client/draw/drawable.cpp [hunk 4]
```diff
@@ -2098,3 +2101,295 @@ void Drawable::Set_Stealth_Look(StealthLookType look)
...
+                        } else {
+                            m_drawableLocoInfo->m_wheelInfo.m_rearLeftHeightOffset =
+                                (0.0f - m_drawableLocoInfo->m_wheelInfo.m_rearLeftHeightOffset) / 2.0f
+                                + m_drawableLocoInfo->m_wheelInfo.m_rearLeftHeightOffset;
+                            m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset =
+                                (0.0f - m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset) / 2.0f
+                                + m_drawableLocoInfo->m_wheelInfo.m_rearRightHeightOffset;
+                        }
+
+                        float major_radius = object->Get_Geometry_Info().Get_Major_Radius();
```
Comment (tomsons26) [inline src/game/client/draw/drawable.cpp]: this is nested too deep

File: src/game/client/draw/drawable.cpp [hunk 5]
```diff
@@ -2098,3 +2101,295 @@ void Drawable::Set_Stealth_Look(StealthLookType look)
...
+                            wheel_info.m_wheelAngle = -wheel_info.m_wheelAngle;
+                        }
+
+                        m_drawableLocoInfo->m_wheelInfo.m_wheelAngle =
+                            (wheel_info.m_wheelAngle - m_drawableLocoInfo->m_wheelInfo.m_wheelAngle) / 10.0f
+                            + m_drawableLocoInfo->m_wheelInfo.m_wheelAngle;
+                        float offset_mult = 0.9f;
+
+                        if (offset1 < 0.0f) {
+                            wheel_info.m_frontLeftHeightOffset = (offset1 / 3.0f + offset1 / 2.0f) * offset_mult;
```
Comment (tomsons26) [inline src/game/client/draw/drawable.cpp]: ifyou are just gonna reuse left then reuse for all where applicable

File: src/game/client/draw/drawable.cpp (hunk 3 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp (hunk 4 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

File: src/game/client/draw/drawable.cpp (hunk 5 above)
Comment (jonwil) [reply src/game/client/draw/drawable.cpp]: fixed.

**Review by xezon — APPROVED** [review]
Wheels and tracks movement in game looks good.

**Pull request merged**